common/configuration/dal/dal.go

```diff
@@ -45,3 +46,35 @@ func (d *DAL) ListModuleConfiguration(ctx context.Context) ([]sql.ModuleConfigur
 	}
 	return l, nil
 }
+
+func (d *DAL) GetModuleSecretURL(ctx context.Context, module optional.Option[string], name string) (string, error) {
+	b, err := d.db.GetModuleSecretURL(ctx, module, name)
+	if err != nil {
+		return "", fmt.Errorf("could not get secret URL: %w", dalerrs.TranslatePGError(err))
+	}
+	return b, nil
+}
+
+func (d *DAL) SetModuleSecretURL(ctx context.Context, module optional.Option[string], name string, url string) error {
+	err := d.db.SetModuleSecretURL(ctx, module, name, url)
+	if err != nil {
+		return fmt.Errorf("could not set secret URL: %w", dalerrs.TranslatePGError(err))
+	}
+	return nil
+}
+
+func (d *DAL) UnsetModuleSecret(ctx context.Context, module optional.Option[string], name string) error {
+	err := d.db.UnsetModuleSecret(ctx, module, name)
+	if err != nil {
+		return fmt.Errorf("could not unset secret: %w", dalerrs.TranslatePGError(err))
+	}
+	return nil
+}
+
+func (d *DAL) ListModuleSecrets(ctx context.Context) ([]sql.ModuleSecret, error) {
```

Comment:
The DAL shouldn't return types from `sql.` directly, that's breaking abstractions. If you want the exact same type though, just do `type ModuleSecret sql.ModuleSecret` and return that.